src/Compilers/VisualBasic/Test/Emit/Emit/EditAndContinue/SymbolMatcherTests.vb

```diff
@@ -512,5 +512,327 @@ End Class
+Public Structure Vector
+    Public Coordinates As (x As Integer, y As Integer, z As Integer)
+End Structure
+"
+            Dim compilation0 = CreateCompilationWithMscorlib(source0, options:=TestOptions.DebugDll, references:=ValueTupleRefs)
+            Dim compilation1 = compilation0.WithSource(source1)
+
+            Dim matcher = New VisualBasicSymbolMatcher(
+                Nothing,
+                compilation1.SourceAssembly,
+                New EmitContext(),
+                compilation0.SourceAssembly,
+                New EmitContext(),
+                Nothing)
+
+            Dim member = compilation1.GetMember(Of FieldSymbol)("Vector.Coordinates")
+            Dim other = matcher.MapDefinition(member)
+            ' If a type changes within a tuple, we do not expect types to match.
+            Assert.Null(other)
+        End Sub
+
+        <Fact>
+        Public Sub TupleStructField_NameChange()
+
+            Dim source0 = "
+Public Structure Vector
+    Public Coordinates As (x As Integer, y As Integer)
+End Structure
+"
+            Dim source1 = "
+Public Structure Vector
+    Public Coordinates As (x As Integer, z As Integer)
+End Structure
+"
+            Dim compilation0 = CreateCompilationWithMscorlib(source0, options:= TestOptions.DebugDll, references:= ValueTupleRefs)
+            Dim compilation1 = compilation0.WithSource(source1)
+
+            Dim matcher = New VisualBasicSymbolMatcher(
+                Nothing,
+                compilation1.SourceAssembly,
+                New EmitContext(),
+                compilation0.SourceAssembly,
+                New EmitContext(),
+                Nothing)
+
+            Dim member = compilation1.GetMember(Of FieldSymbol)("Vector.Coordinates")
+            Dim other = matcher.MapDefinition(member)
+            ' Types must match because just an element name was changed.
+            Assert.NotNull(other)
+        End Sub
+
+        <Fact>
+        Public Sub TupleDelegate_TypeChange()
+
+            Dim source0 = "
+Public Class C
+    Public Delegate Function F() As (Integer, Integer)
+End Class
+"
+            Dim source1 = "
+Public Class C
+    Public Delegate Function F() As (Integer, Boolean)
+End Class
+"
+            Dim compilation0 = CreateCompilationWithMscorlib(source0, options:= TestOptions.DebugDll, references:= ValueTupleRefs)
+            Dim compilation1 = compilation0.WithSource(source1)
+
+            Dim matcher = New VisualBasicSymbolMatcher(
+                Nothing,
+                compilation1.SourceAssembly,
+                New EmitContext(),
+                compilation0.SourceAssembly,
+                New EmitContext(),
+                Nothing)
+
+            Dim member = compilation1.GetMember(Of SourceNamedTypeSymbol)("C.F")
+            Dim other = matcher.MapDefinition(member)
+            ' Tuple delegate defines a type. We should be able to match old and new types by name.
+            Assert.NotNull(other)
+        End Sub
```

Comment:
Thank you! Added

---
In reply to: [117565093](https://github.com/dotnet/roslyn/pull/19283#discussion_r117565093) [](ancestors = 117565093)